MPF.Core/Modules/XboxBackupCreator/Parameters.cs

```diff
@@ -0,0 +1,684 @@
+                return null;
+
+            // Example:
+            // ========================================
+            // < --Security Sector Details -->
+            // Source Drive: SH-D162D
+            // ----------------------------------------
+
+            try
+            {
+                // Parse drive model from log file
+                using var sr = File.OpenText(log);
+                while (!sr.EndOfStream)
+                {
+                    string? line = sr.ReadLine()?.Trim();
+                    if (line?.StartsWith("Source Drive: ") == true)
+                    {
+                        return line.Substring("Source Drive: ".Length).Trim();
+                    }
+                }
+
+                // We couldn't detect the drive model
+                return null;
+            }
+            catch
+            {
+                // We don't care what the exception is right now
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Get the Layerbreak value if possible
+        /// </summary>
+        /// <param name="dvd">Path to layerbreak file</param>
+        /// <param name="layerbreak">Layerbreak value if found</param>
+        /// <returns>True if successful, otherwise false</returns>
+        /// <returns></returns>
+        private static bool GetLayerbreak(string? dvd, out long layerbreak)
+        {
+            layerbreak = 0;
+
+            if (string.IsNullOrEmpty(dvd) || !File.Exists(dvd))
+                return false;
+
+            // Example:
+            // LayerBreak=1913776
+            // track.iso
+
+            try
+            {
+                // Parse Layerbreak value from DVD file
+                using var sr = File.OpenText(dvd);
+                while (!sr.EndOfStream)
+                {
+                    string? line = sr.ReadLine()?.Trim();
+                    if (line?.StartsWith("LayerBreak=") == true)
+                    {
+                        return long.TryParse(line.Substring("LayerBreak=".Length).Trim(), out layerbreak);
+                    }
+                }
+
+                // We couldn't detect the Layerbreak
+                return false;
+            }
+            catch
+            {
+                // We don't care what the exception is right now
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Get the read error count if possible
+        /// </summary>
+        /// <param name="log">Path to XBC log file</param>
+        /// <param name="readErrors">Read error count if found, -1 otherwise</param>
+        /// <returns>True if sucessful, otherwise false</returns>
+        private bool GetReadErrors(string? log, out long readErrors)
+        {
+            readErrors = -1;
+
+            if (string.IsNullOrEmpty(log) || !File.Exists(log))
+                return false;
+
+            // TODO: Logic when more than one dump is in the logs
```

Comment:
I think it's reasonable (for now) to have it only read the first instance.

Reply:
I was thinking people would forget to delete the Log, then dump, then copy the log file. The good dump ends up being at the end..

Reply:
I'm guessing you saw some examples of this when researching?

Reply:
Just my own :)

Reply:
I'll take it for now.